src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -193,12 +193,20 @@ public void init(AdaptorContext context) throws Exception {
       log.config("adaptor runs in lister-only mode");
     }
 
-    modeOfOperation = cfg.getValue("db.modeOfOperation");
+    modeOfOperation = cfg.getValue("db.modeOfOperation").trim();
     if ("urlAndMetadataLister".equals(modeOfOperation) && encodeDocId) {
       String errmsg = "db.modeOfOperation of \"" + modeOfOperation
           + "\" requires docId.isUrl to be \"true\"";
       throw new InvalidConfigurationException(errmsg);
     }
+    String modeOfOperationValid[] = {"blobColumn", "filepathColumn",
```

Comment:
Can you please clarify what do you mean for "use the same reflection mechanism that is used to do the work"?
If you prefer I can just remove this check. I considered it a simple check - based on the bug title - to avoid misspelled values.